server/container/guice/guice-utils/src/main/java/org/apache/james/utils/ExtensionConfiguration.java

```diff
@@ -19,24 +19,25 @@
 
 package org.apache.james.utils;
 
-import java.util.Arrays;
 import java.util.List;
-import java.util.Optional;
 
 import org.apache.commons.configuration2.Configuration;
 
 import com.github.steveash.guavate.Guavate;
+import com.google.common.base.Splitter;
 import com.google.common.collect.ImmutableList;
 
 public class ExtensionConfiguration {
     public static final ExtensionConfiguration DEFAULT = new ExtensionConfiguration(ImmutableList.of());
 
     public static ExtensionConfiguration from(Configuration configuration) {
-        List<String> list = Optional.ofNullable(configuration.getStringArray("guice.extension.module"))
-            .map(Arrays::asList)
-            .orElse(ImmutableList.of());
+        String rawString = configuration.getString("guice.extension.module", "");
 
-        return new ExtensionConfiguration(list.stream()
+        return new ExtensionConfiguration(Splitter.on(",")
+            .omitEmptyStrings()
+            .trimResults()
```

Comment:
maybe extract 
```
Splitter.on(",")
            .omitEmptyStrings()
            .trimResults()
 ```
to a local constant ? I don't think we need to build a new instance for every call (not that it matters much since this should only run once per process)

Reply:
This code path is only executed once.

Do we care? Sounds like early optimisation to me...

Reply:
I feel that the splitter setup gets in the way of the actual parsing logic. 
The first 3 lines are type Splitter, then the flow changes to stream processing with the actual input value only appearing on line 4 

extracting to a constant would change the code to 
```
SPLITTER.splitToList(rawString)
            .stream()
            .map(ClassName::new)
            .collect(Guavate.toImmutableList())
```
An alternative could be extracting to a private method that event takes care of the list<->stream ceremony  

``` 
splitOnCommas(rawString)
 .map(ClassName::new)
            .collect(Guavate.toImmutableList())
```

Reply:
I had a look, Joiner and Splitter are thread safe, thus can be safely shared as constants

The point is we never did it in the source code. I think it could be nice to conduct a global refactoring regarding this.

I guess the best way to move is thus fix these usage here and open another PR for doing this refctoring.